app/V1Module/security/ACLModule.php

```diff
@@ -6,19 +6,23 @@
   /** @var UserStorage */
   private $userStorage;
 
+  /** @var Identity */
+  private $identity;
+
   /** @var IAuthorizator */
   private $authorizator;
 
-  public function __construct(UserStorage $userStorage, IAuthorizator $authorizator) {
+  public function __construct(UserStorage $userStorage, IAuthorizator $authorizator, ?Identity $identity = null) {
     $this->userStorage = $userStorage;
     $this->authorizator = $authorizator;
+    $this->identity = $identity;
   }
 
   protected abstract function getResourceName();
 
   protected function check($action, $context): bool {
     return $this->authorizator->isAllowed(
-      $this->userStorage->getIdentity(),
+      $this->identity ?? $this->userStorage->getIdentity(),
```

Comment:
Would not it be better to set the identity (from arg or from userStorage) in the constructor?